tests/test_crawler.py

```diff
@@ -305,3 +305,21 @@ def test_asyncio_enabled_reactor(self):
         log = self.run_script('asyncio_enabled_reactor.py')
         self.assertIn('Spider closed (finished)', log)
         self.assertIn("DEBUG: Asyncio reactor is installed", log)
+
+    def test_ipv6_default_name_resolver(self):
+        log = self.run_script('default_name_resolver.py')
+        self.assertIn('Spider closed (finished)', log)
+        self.assertIn("twisted.internet.error.DNSLookupError: DNS lookup failed: no results for hostname lookup: ::1.", log)
+        self.assertIn("'downloader/exception_type_count/twisted.internet.error.DNSLookupError': 1,", log)
+
+    def test_ipv6_alternative_name_resolver(self):
+        log = self.run_script('alternative_name_resolver.py')
+        self.assertIn('Spider closed (finished)', log)
+        self.assertTrue(any([
+            "twisted.internet.error.ConnectionRefusedError" in log,
```

Comment:
Locally I get `ConnectionRefusedError`, but `ConnectionRefusedError` in the Travis environment.

Reply:
We test different twisted versions, I guess it is acceptable.